lightbar_manager/src/lightbar_manager/lightbar_manager_node.cpp

```diff
@@ -0,0 +1,320 @@
+{
+    // Since lightbar_manager is guaranteed to have the highest priority, the manager is guaranteed to get all the control upon request
+    std::vector<LightBarIndicator> all_indicators, denied_list;
+
+    ROS_INFO_STREAM("LightBarManager was commanded to turn off all indicators!");
+
+    for (std::pair <LightBarIndicator, std::string> element : lbm_.getIndicatorControllers())
+        all_indicators.push_back(element.first);
+    
+    denied_list = lbm_.requestControl(all_indicators, node_name_);
+
+    if (denied_list.size() != 0 )
+        ROS_WARN_STREAM ("In Function " << __FUNCTION__ << ": LightBarManager was not able to take all indicator's control");
+
+    int response_code = 0;
+    for (auto indicator : all_indicators)
+    {
+        response_code = setIndicator(indicator, OFF, node_name_);
+        if (response_code != 0)
+            ROS_WARN_STREAM ("In Function " << __FUNCTION__ << ": LightBarManager was not able to turn off indicator ID:" 
+                << indicator << ". Response code: " << response_code);
+    }
+    
+    // Once forced them off, release controls in case carma still is running; otherwise, lightbar_manager will hog the control
+    all_indicators.erase(std::remove(all_indicators.begin(), all_indicators.end(), GREEN_FLASH), all_indicators.end());
+    all_indicators.erase(std::remove(all_indicators.begin(), all_indicators.end(), GREEN_SOLID), all_indicators.end());
+    lbm_.releaseControl(all_indicators, node_name_);
+}
+
+bool LightBarManager::requestControlCallBack(cav_srvs::RequestIndicatorControlRequest& req, cav_srvs::RequestIndicatorControlResponse& res)
+{
+    std::vector<LightBarIndicator> ind_list, controlled_ind_list;
+    std::vector<LightBarCDAType> controlled_cda_type_list;
+
+    // Use CDAType if the field is not empty in the request
+    if (req.cda_list.size() != 0)
+    {   
+        for (auto cda_type : req.cda_list) 
+        {
+            // return false if invalid cda_type number
+            if (static_cast<int>(cda_type.type) >= INDICATOR_COUNT)
+                return false; 
+            ind_list.push_back(lbm_.getIndicatorFromCDAType(static_cast<LightBarCDAType>(cda_type.type)));
+        }
+    }
+    else
+    {
+        for (auto indicator : req.ind_list) 
+        {
+            // return false if invalid indicator number
+            if (static_cast<int>(indicator.indicator) >= INDICATOR_COUNT)
+                return false;
+            ind_list.push_back(static_cast<LightBarIndicator>(indicator.indicator));
+        }
+            
+    }
+    
+    controlled_ind_list = lbm_.requestControl(ind_list, req.requester_name);
+    
+    for (auto indicator : controlled_ind_list)
+    {
+        try
+        {
+            // Some indicators do not have mapped CDAType
+            controlled_cda_type_list.push_back(lbm_.getCDATypeFromIndicator(indicator));
+        }
+        catch(INDICATOR_NOT_MAPPED)
+        {
+            continue;
```

Comment:
Should we log a warning here? Or is it expected for this to occur regularly?

Reply:
Since many of them doesn't have mapping, and left_arrow and right_arrow out of those, for example, are expected to be used a lot I am guessing. So I was thinking of not logging it.